test/python/compiler/test_assembler.py

```diff
@@ -401,12 +401,29 @@ def test_assemble_meas_map(self):
 
     def test_assemble_memory_slots(self):
         """Test assembling a schedule and inferring number of memoryslots."""
+        acquire_channels = [pulse.AcquireChannel(idx) for idx in range(10)]
+        mem_slots = [pulse.MemorySlot(idx) for idx in range(10)]
+
         acquire = pulse.Acquire(5)
-        n_memoryslots = 10
-        schedule = acquire(self.device.acquires[0], mem_slots=pulse.MemorySlot(n_memoryslots))
 
-        qobj = assemble(schedule, meas_map=[[0], [1]])
-        self.assertEqual(qobj.config.memory_slots, n_memoryslots)
+        schedule = pulse.Schedule(name='all_acquire')
+        schedule = schedule.insert(0, acquire(acquire_channels, mem_slots))
+
+        qobj = assemble(schedule)
+        test_dict = qobj.to_dict()
+
+        self.assertEqual(test_dict['config']['memory_slots'], 10)
+
+        schedule = pulse.Schedule(name='partial_acquire')
+        schedule = schedule.insert(0, acquire([acquire_channels[3], acquire_channels[5]],
+                                              [mem_slots[3], mem_slots[5]]))
+        schedule = schedule.insert(0, acquire([acquire_channels[7]],
+                                              [mem_slots[7]]))
+
+        qobj = assemble(schedule)
+        test_dict = qobj.to_dict()
```

Comment:
There is no need to convert this to a dictionary. Just access the `memory_slots` attribute.